test/e2e/storage/external/external.go

```diff
@@ -0,0 +1,270 @@
+	}
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	// Some reasonable defaults follow.
+	driver := &driverDefinition{
+		DriverInfo: testsuites.DriverInfo{
+			SupportedFsType: sets.NewString(
+				"", // Default fsType
+			),
+		},
+		ClaimSize: "5Gi",
+	}
+	// TODO: strict checking of the file content once https://github.com/kubernetes/kubernetes/pull/71589
+	// or something similar is merged.
+	if err := runtime.DecodeInto(legacyscheme.Codecs.UniversalDecoder(), data, driver); err != nil {
+		return nil, errors.Wrap(err, filename)
+	}
+	return driver, nil
+}
+
+var _ testsuites.TestDriver = &driverDefinition{}
+
+// We have to implement the interface because dynamic PV may or may
+// not be supported. driverDefinition.SkipUnsupportedTest checks that
+// based on the actual driver definition.
+var _ testsuites.DynamicPVTestDriver = &driverDefinition{}
+
+// Same for snapshotting.
+var _ testsuites.SnapshottableTestDriver = &driverDefinition{}
+
+// runtime.DecodeInto needs a runtime.Object but doesn't do any
+// deserialization of it and therefore none of the methods below need
+// an implementation.
+var _ runtime.Object = &driverDefinition{}
+
+// DriverDefinition needs to be filled in via a .yaml or .json
+// file. It's methods then implement the TestDriver interface, using
+// nothing but the information in this struct.
+type driverDefinition struct {
+	// DriverInfo is the static information that the storage testsuite
+	// expects from a test driver. See test/e2e/storage/testsuites/testdriver.go
+	// for details. The only field with a non-zero default is the list of
+	// supported file systems (SupportedFsType): it is set so that tests using
+	// the default file system are enabled.
+	DriverInfo testsuites.DriverInfo
+
+	// ShortName is used to create unique names for test cases and test resources.
+	ShortName string
+
+	// StorageClass must be set to enable dynamic provisioning tests.
+	// The default is to not run those tests.
+	StorageClass struct {
+		// FromName set to true enables the usage of a
+		// storage class with DriverInfo.Name as provisioner.
+		FromName bool
+
+		// FromFile is used only when FromName is false.
+		// It loads a storage class from the given .yaml or .json
+		// file. File names are resolved by the framework.testfiles
+		// package, which typically means that they can be absolute or
+		// relative to the test suite's --repo-root parameter.
+		FromFile string
+	}
+
+	// SnapshotClass must be set to enable snapshotting tests.
+	// The default is to not run those tests.
+	SnapshotClass struct {
+		// FromName set to true enables the usage of a
+		// snapshotter class with DriverInfo.Name as provisioner.
+		FromName bool
+
+		// TODO (?): load from file
+	}
+
+	// ClaimSize defines the desired size of dynamically
+	// provisioned volumes. Default is "5GiB".
+	ClaimSize string
+
+	// ClientNodeName selects a specific node for scheduling test pods.
```

Comment:
Clarify this is only needed for drivers like hostpath AND do not support topology

Reply:
We do need this option right now ourselves for testing the hostpath driver as part of the upcoming kubernetes-csi CI. It's okay to explain that it shouldn't be needed and I have added a comment about it, but we should still have that option.